### PR TITLE
Layout Improvements

### PR DESCRIPTION
**Pull request by MitchellNZ**
More details can be found at #1089 

@stsrki I have added a `LoadingClass` to `Layout` to represent what you suggested on #1084.
This can be overridden with CSS for a spinner or progress bar, but I have also provided a default.

Let me know what you think, didn't want to spend too much time on that right now.

### CODE REVIEW

File: Source/Blazorise/Layout.razor [hunk 1]
```diff
@@ -3,6 +3,10 @@
 {
     <section @ref="@ElementRef" id="@ElementId" class="@ClassNames" style="@StyleNames" @attributes="@Attributes">
         <CascadingValue Value=this>
+            @if ( Loading )
+            {
+                <div class="@LoadingClass"></div>
```
Comment (stsrki) [inline Source/Blazorise/Layout.razor:8]: I was thinking of adding additional `[Parameter] public RenderFragment LoadingTemplate { get; set; }`, and then add it like:

```cshtml
@if ( LoadingTemplate != null )
{
    @LoadingTemplate
}
else
{
    <div class="@LoadingClass"></div>
}
```

File: Source/Blazorise/Layout.razor.cs [hunk 2]
```diff
@@ -14,7 +14,9 @@ public partial class Layout : BaseComponent
 
         private bool sider;
 
-        private bool bodyClassApplied;
+        private bool loading;
+
+        private string loadingClass = "b-layout-loading-spinner";
```
Comment (stsrki) [inline Source/Blazorise/Layout.razor.cs:19]: Make this as ClassBuilder.

File: Source/Blazorise/Layout.razor.cs [hunk 3]
```diff
@@ -47,6 +51,32 @@ public bool Sider
...
+            set
+            {
+                loading = value;
+
+                DirtyClasses();
+            }
+        }
+
+        [Parameter]
+        public string LoadingClass
```
Comment (stsrki) [inline Source/Blazorise/Layout.razor.cs:67]: Make LoadingClass as custom class to be added into loading ClassBuilder

File: docs/_docs/components/bar.md [hunk 4]
```diff
@@ -59,8 +59,8 @@ The bar component has the following structure:
     </BarStart>
     <BarEnd>
       <BarItem>
-        <Button Color="Color.Primary">Sign up</Button>
-        <Button Color="Color.Secondary">Log in</Button>
+        <button Color="Color.Primary">Sign up</button>
```
Comment (stsrki) [inline docs/_docs/components/bar.md:62]: Why `button` instead of `Button`? Is this a mistake?

Comment (MitchellNZ) [reply docs/_docs/components/bar.md:62]: Oops! Yes this is a mistake.
Must've auto save formatting

File: Source/Blazorise/Layout.razor (hunk 1 above)
Comment (MitchellNZ) [reply Source/Blazorise/Layout.razor:8]: Ahh I see, looks good!

File: Source/Blazorise/Layout.razor.cs (hunk 2 above)
Comment (MitchellNZ) [reply Source/Blazorise/Layout.razor.cs:19]: Will do, was just doing this quickly to get feedback on templating idea :)

**Comment by stsrki** [comment]
Test and working like a charm. Excellent work!

**Pull request merged**